test/Templates.UnitTests/Templates.UnitTests.csproj

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'Common.props'))\Common.props" />
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <MinimumVisualStudioVersion>11.0</MinimumVisualStudioVersion>
+    <ProjectGuid>{7A328A78-533E-43F8-9AF8-9332A3331418}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <DefaultLanguage>en-US</DefaultLanguage>
+    <TargetFramework>netstandard1.6</TargetFramework>
+    <PackageTargetFallback>dotnet5.4;portable-net451+win8</PackageTargetFallback>
+    <OutDir>$(OutDir)Tests\</OutDir>
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <NonShipping>true</NonShipping>
+    <StartAction>Program</StartAction>
+    <StartProgram>$(DotNetTool).exe</StartProgram>
+    <StartArguments>"$(OutDir)xunit.console.netcore.exe" "$(OutDir)$(AssemblyName).dll" -xml "$(OutDir)TemplatesTestResults.xml" -wait</StartArguments>
+    <StartWorkingDirectory>$(OutDir)</StartWorkingDirectory>
+    <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
+  </PropertyGroup>
+  <PropertyGroup>
+    <FullPathOutDir>$([System.IO.Path]::Combine($(MSBuildProjectDirectory), $(OutDir)))</FullPathOutDir>
+    <PostBuildEvent>
+      xcopy /S /Y /R  "$(MSBuildProjectDirectory)\..\..\src\Templates\*" "$(FullPathOutDir)\TestAssets\Templates\" /EXCLUDE:$(MSBuildProjectDirectory)\..\..\.gitignore
```

Comment:
Can we instead just get to the path of these files in source control and save this copy?

Does xcopy actually understand the full syntax of .gitignore or are we just lucky enough that it respects enough to exclude what we don't want to copy from the templates folder?